### PR TITLE
upgrade core LTS to 2.346.1 and associated bom, fix test due to this upgrade

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

### CODE REVIEW

**Comment by olamy** [comment]
I have created the label `it-works-on-my-machine` (^&%&%$(&**&(%&* :))

**Comment by olamy** [comment]
`interesting` CI fail because the record logs are different in Java 8 and Java 11.

```
Java 8
[
Generating App Installation Token for app ID 54321, 
Generating App Installation Token for app ID 54321, 
Failed to generate new GitHub App Installation Token for app ID 54321: cached token is stale but has not expired, 
Generating App Installation Token for app ID 54321, 
Failed to generate new GitHub App Installation Token for app ID 54321: cached token is stale but has not expired, 
Generating App Installation Token for app ID 54321, 
Generating App Installation Token for app ID 54321]


Java 11
[Generating App Installation Token for app ID 54321, 
Generating App Installation Token for app ID 54321, 
Failed to generate new GitHub App Installation Token for app ID 54321: cached token is stale but has not expired, 
Generating App Installation Token for app ID 54321, 
Failed to generate new GitHub App Installation Token for app ID 54321: cached token is stale but has not expired, 
Generating App Installation Token for app ID 54321]
```

Java 8 has one more line.

**Comment by olamy** [comment]
I'm tempted to ignore Java 8 as anyway Jenkins has already moved to Java 11.

**Comment by olamy** [comment]
this is just flaky with Java 8....

**Comment by olamy** [comment]
now doesn't work on windows with Java 8....

File: pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
         <changelist>999999-SNAPSHOT</changelist>
```
Comment (Mato220) [inline pom.xml:24]: <changelist>jenkins>SNAPSHOT</changelist>

**Review by Mato220 — CHANGES_REQUESTED** [review]

**Comment by jtnord** [comment]
so is it time to declare the test flaky yet or does it reliably fail somewhere?

**Comment by olamy** [comment]
> so is it time to declare the test flaky yet or does it reliably fail somewhere?

with the upgrade it's stable with linux/java11 and windows/java8.
but really flaky with linux/java8...

As we will probably move soon to java11 this will not be a problem. 
maybe we can simply skip this test for linux/java8 until we upgrade?

File: src/test/java/org/jenkinsci/plugins/github_branch_source/GithubAppCredentialsTest.java [hunk 2]
```diff
@@ -455,41 +463,45 @@ public void testAgentRefresh() throws Exception {
...
-              // checkout scm
-              // (No token generation)
-              // sleep
               // checkout scm
               "Generating App Installation Token for app ID 54321",
               // (error forced by wiremock)
               "Failed to generate new GitHub App Installation Token for app ID 54321: cached token is stale but has not expired",
               // (error forced by wiremock - failed refresh on the agent)
               // "Generating App Installation Token for app ID 54321 on agent", // 1
-              "Generating App Installation Token for app ID 54321 for agent",
```
Comment (jtnord) [inline src/test/java/org/jenkinsci/plugins/github_branch_source/GithubAppCredentialsTest.java:473]: Ahh..

@Olamy this is incorrect - the test was failing due to JENKINS-69488 which is why my PR is breaking this test :)

Comment (olamy) [reply src/test/java/org/jenkinsci/plugins/github_branch_source/GithubAppCredentialsTest.java:473]: good catch

**Pull request merged**